test/sys/test_wait.rs

```diff
@@ -34,3 +34,56 @@ fn test_wait_exit() {
       Err(_) => panic!("Error: Fork Failed")
     }
 }
+
+#[cfg(any(target_os = "linux", target_os = "android"))]
+// FIXME: qemu-user doesn't implement ptrace on most arches
+#[cfg_attr(not(any(target_arch = "x86", target_arch = "x86_64")), ignore)]
+mod ptrace {
+    use nix::Result;
+    use nix::sys::ptrace::*;
+    use nix::sys::ptrace::ptrace::*;
+    use nix::sys::signal::*;
+    use nix::sys::wait::*;
+    use nix::unistd::*;
+    use nix::unistd::ForkResult::*;
+    use libc::_exit;
+    use std::ptr;
+
+    fn ptrace_child() -> Result<()> {
+        try!(ptrace(PTRACE_TRACEME, Pid::from_raw(0), ptr::null_mut(), ptr::null_mut()));
+        // As recommended by ptrace(2), raise SIGTRAP to pause the child
+        // until the parent is ready to continue
+        try!(raise(SIGTRAP));
+        unsafe {_exit(0)}
```

Comment:
It's not necessary to skip all the C-level cleanup stuff with `_exit(0)`.  Instead, you can simply use `std::process::exit(0)` and eliminate the `unsafe` block.